lib/puppet/provider/file_line/ruby.rb

```diff
@@ -43,7 +52,7 @@ def handle_create_with_match()
 
     File.open(resource[:path], 'w') do |fh|
       lines.each do |l|
-        fh.puts(regex.match(l) ? resource[:line] : l)
+        fh.puts(match_regex.match(l) ? resource[:line] : l)
```

Comment:
it also isn't a new behavior with this patch. The line only changes because where the `Regexp` is created is moved and renamed from `regex` to `match_regex`